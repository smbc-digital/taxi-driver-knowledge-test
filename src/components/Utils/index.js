import 'isomorphic-fetch'
import moment from 'moment'

export const formatAvailableAppointments = appointments => {
	return appointments.map(appointment => {
		return {
			date: appointment.date,
			displayDate: moment(appointment.date, 'DD/MM/YYYY').format('dddd D MMMM YYYY'),
			times: appointment.times.map(time => {
				const value = moment(`${appointment.date} ${time.startTime}`, 'DD/MM/YYYY HH:mm:ss').format('YYYY-MM-DD HH:mm:ss')
				return {
					startTime: time.startTime,
					id: value,
					value,
					label: moment(time.startTime, 'HH:mm:ss').format('H:mma'),
					name: 'testDate'
				}
			})
		}
	})
}

export const getAvailableAppointments = async (isResit, from, to) => {
	const request = {
		 from: from,
		 to: to,
		 isResit: isResit
	}

	try {
		const result = await fetch('/book-taxi-driver-knowledge-test/available-appointments', {
			method: 'POST',
			headers: {
				Accept: 'application/json; charset=utf-8',
				'Content-Type': 'application/json; charset=utf-8'
			},
			body: JSON.stringify(request)
		})

		const responseObject = await result.json()

		return {
			status: result.status,
			appointments: formatAvailableAppointments(responseObject)
		}
	} catch (error) {
		return {
			status: 500,
			appointments: []
		}
	}
}

export const reserveAppointment = async ({isResit, testDate, firstName, lastName, phoneNumber, emailAddress, address, testType, previousTestDate}) => {
	try {
		const request = {
			testDate, 
			isResit,
			name: `${firstName.value} ${lastName.value}`,
			address: address.value.selectedAddress || `${address.value.addressLine1} ${address.value.addressLine2} ${address.value.town} ${address.value.postcode}`,
			email: emailAddress.value,
			phoneNumber: phoneNumber.value,
			testType: testType.value,
			previousTestDate: previousTestDate.value
		}

		const result = await fetch('/book-taxi-driver-knowledge-test/pencil-an-appointment', {
			method: 'POST',
			headers: {
				Accept: 'application/json; charset=utf-8',
				'Content-Type': 'application/json; charset=utf-8'
			},
			body: JSON.stringify(request)
		})

		const responseObject = await result.json()

		return {
			status: result.status,
			bookingId: responseObject
		}
	} 
	catch (error) {
		return {
			status: 500
		}
	}
}

export const getPaymentUrl = async (bookingId, testDate) => {
	try {
		const result = await fetch('/book-taxi-driver-knowledge-test/generate-payment-url', {
				method: 'POST',
				headers: {
					Accept: 'application/json; charset=utf-8',
					'Content-Type': 'application/json; charset=utf-8'
				},
				body: JSON.stringify({
					bookingId, testDate
				})
			})

		const url = await result.text()

		return {
			status: result.status,
			url
		}
	} catch (error) {
		return {
			status: 500
		}
	}
}
