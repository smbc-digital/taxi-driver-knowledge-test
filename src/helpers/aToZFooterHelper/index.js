const showAToZFooter = (show) =>{
    let footer  = document.getElementById('atoz')
    if(footer !== null) 
    {
        if(show){
            footer.classList.remove('hidden')
        }
        else{
            footer.classList.add('hidden')
        }
    }
}

export default showAToZFooter