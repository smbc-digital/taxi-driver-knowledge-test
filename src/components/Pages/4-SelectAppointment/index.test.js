import { React, mount } from '../../../helpers/SetupTest'
import { SelectAppointment } from './index'
// import { getPageRoute } from '../../../helpers/pagehelper'
import * as submitUtils from '../../Utils'
// import moment from 'moment'

describe('SelectAppointment', () => {
	it('should call push on submit', () => {
		//Arrange
		const data = {
			isResit: {
				value: false,
				isValid: true
			},
			previousTestDate: {
				value: '',
				isValid: false
			},
			testDate: {
				value: '25/04/2019',
				isValid: true
			}
		}

		const history = { push: jest.fn() }
		submitUtils.reserveAppointment = jest.fn().mockReturnValue({ status: 200, response: {bookingId: 'any'} })
		const wrapper = mount(<SelectAppointment context={data} history={history} />)

		//Act
		wrapper.find('form').simulate('submit')

		//Assert
		// expect(history.push).toHaveBeenCalledWith(getPageRoute(5))
	})


	// it('should update state when showMore clicked', () => {
	// 	//Arrange
	// 	const data = {
	// 		appointments: [
	// 			{
	// 				date: '27/02/2019',
	// 				times: [
	// 					{
	// 						startTime: '08:45:00',
	// 						endTime: '11:00:00'
	// 					},
	// 					{
	// 						startTime: '12:45:00',
	// 						endTime: '15:00:00'
	// 					}
	// 				]
	// 			},
	// 			{
	// 				date: '06/03/2019',
	// 				times: [
	// 					{
	// 						startTime: '08:45:00',
	// 						endTime: '11:00:00'
	// 					}
	// 				]
	// 			}
	// 		],
	// 		eighteenWeekAppointments: [
	// 			{
	// 				date: 'Wednesday 27 February 2019',
	// 				times: [
	// 					{
	// 						startTime: '8:45am',
	// 						endTime: '11:00:00'
	// 					},
	// 					{
	// 						startTime: '12:45pm',
	// 						endTime: '15:00:00'
	// 					}
	// 				]
	// 			},
	// 			{
	// 				date: 'Wednesday 6 March 2019',
	// 				times: [
	// 					{
	// 						startTime: '8:45am',
	// 						endTime: '11:00:00'
	// 					}
	// 				]
	// 			},
	// 			{
	// 				date: 'Wednesday 13 March 2019',
	// 				times: [
	// 					{
	// 						startTime: '8:45am',
	// 						endTime: '11:00:00'
	// 					},
	// 					{
	// 						startTime: '12:45pm',
	// 						endTime: '15:00:00'
	// 					}
	// 				]
	// 			}
	// 		]
	// 	}

	// 	const history = { push: jest.fn() }
	// 	const event = { preventDefault: jest.fn() }
	// 	const wrapper = mount(<SelectAppointment context={data} history={history} />)

	// // 	//Act
	// 	wrapper.instance().onClick(event)

	// 	//Assert
	// 	expect(wrapper.state().showMore).toEqual(false)
	// })

	// it('on componentDidMount did update state', async () => {
	// 	const data = {
	// 		isResit: {
	// 			value: false,
	// 			isValid: false
    //         },
    //         previousTestDate: {
	// 			value: '26/02/2019',
	// 			isValid: false
	// 		},
	// 		appointments: [],
	// 		eighteenWeekAppointments: [],
	// 		twelveWeekAppointments : []
	// 	}

	// 	const history = { push: jest.fn() }
	// 	const wrapper = mount(<SelectAppointment context={data} history={history} />)

	// 	//Act
	// 	wrapper.instance().componentDidMount()

	// 	expect(wrapper.state().appointments).toEqual([])
	// })
})
