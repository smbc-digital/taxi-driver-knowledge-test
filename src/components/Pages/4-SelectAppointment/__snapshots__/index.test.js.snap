// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectAppointment snapshot renders correctly with spinner while there are no appointments 1`] = `
Array [
  <form
    onSubmit={[Function]}
  >
    <h1 />
    <div>
      <section
        className="alert-information-form"
      >
        <span
          className="fa fa-info-circle"
        />
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": "The test usually takes place on a Wednesday. There may be times when we run the test on other days, these will be displayed below.",
            }
          }
        />
      </section>
    </div>
    <h2>
      Select an appointment
    </h2>
    <div
      className="page-loading-spinner"
    />
    <a
      className="anchor-with-icon"
      href="#"
      onClick={[Function]}
    >
       
      <i
        className="fa fa-plus-square fa-lg"
      />
      Show more dates
    </a>
    <button
      className="button-primary"
      disabled={true}
      id={undefined}
      onClick={undefined}
      type="submit"
    >
      Next step
    </button>
  </form>,
  <p>
    <a
      href="#"
      onClick={[Function]}
    >
      Previous
    </a>
  </p>,
]
`;
